receiver/nvmlreceiver/client.go

```diff
@@ -278,6 +321,93 @@ func (client *nvmlClient) collectDeviceMemoryInfo() []nvmlMetric {
 	return deviceMetrics
 }
 
+func (client *nvmlClient) collectProcessMetrics() []processMetric {
+	if client.disable || !client.collectProcessInfo {
+		return nil
+	}
+
+	processMetrics := make([]processMetric, 0)
+
+	for gpuIndex, device := range client.devices {
+		pids, ret := nvmlDeviceGetAccountingPids(device)
+		if ret != nvml.SUCCESS {
+			msg := fmt.Sprintf("Unable to query cached PIDs on '%v", nvml.ErrorString(ret))
+			client.issueWarningForFailedQueryUptoThreshold(gpuIndex, "nvml.processes", msg)
+			continue
+		}
+
+		for _, pid := range pids {
+			metricName := fmt.Sprintf("nvml.processes{pid=%d}", pid)
+
+			stats, ret := nvml.DeviceGetAccountingStats(device, uint32(pid))
+			if ret != nvml.SUCCESS {
+				msg := fmt.Sprintf("Unable to query pid %d account statistics on '%v", pid, nvml.ErrorString(ret))
+				client.issueWarningForFailedQueryUptoThreshold(gpuIndex, metricName, msg)
+				continue
+			}
+
+			startTime := time.UnixMicro(int64(stats.StartTime))
+			if stats.IsRunning != 1 && startTime.Before(client.lastProcessInfoCollection) {
```

Comment:
@LujieDuan fyi

Reply:
I can try to confirm the behavior by running ephemeral jobs and seeing what it emits.

Reply:
![image](https://user-images.githubusercontent.com/4366369/211116789-73cbe486-1435-4390-9946-aeeb4ac634a2.png)

Wdyt? Remove it? Might not leave a good impression. I say drop it and mention in the documentation that process needs to be alive at the time of capture.

Reply:
Though I haven't seen it in practice so far, this does raise the prospect of a race condition on which isRunning was set by NVIDIA and the process is dead by the time it's metadata is queried.

Reply:
Yes SGTM - let's drop those and have this documented. Thanks for testing this!

Reply:
Another concern this might raise is for a hypothetical inference scenario. A typical inference could run from 20-100 ms. If each inference request is served on a separate process (seems far-fetched, but maybe?) it could generate up to 1000 ephemeral processes. At that point NVIDIA's accounting statistics buffers would be full. And we'd end up reporting X metrics every collection, where X is NVIDIA's ring buffer size.

Reply:
Yeah that's a good point. I think it is a good idea to skip emitting deceased process metrics for now. So even if we are reporting the last X metrics, we have all the labels set correctly. 
For the race condition - I don't see a good way to avoid it if we choose to attach those process labels. We could document that and if it ever happens I believe the users can still group the metrics by PID when viewing the metrics.

Reply:
Sg to me.